src/test/ui/destructuring-assignment/tuple_destructure.rs

```diff
@@ -0,0 +1,19 @@
+// run-pass
+
+#![feature(destructuring_assignment)]
+
+fn main() {
+    let (mut a, mut b);
+    (a, b) = (0, 1);
+    assert_eq!((a, b), (0, 1));
+    (b, a) = (a, b);
```

Comment:
please also add a test for `!Copy` things like `String`

Reply:
Added :)